backend/gateway/mux/mux.go

```diff
@@ -201,24 +204,14 @@ func customHeaderMatcher(key string) (string, bool) {
 }
 
 func customErrorHandler(ctx context.Context, mux *runtime.ServeMux, m runtime.Marshaler, w http.ResponseWriter, req *http.Request, err error) {
-	//  TODO(maybe): once we have non-browser clients we probably want to avoid the redirect and directly return the error.
-	if s, ok := status.FromError(err); ok && s.Code() == codes.Unauthenticated {
-		referer := req.Referer()
-		redirectPath := "/v1/authn/login"
-		if len(referer) != 0 {
-			referer, err := url.Parse(referer)
-			if err != nil {
-				runtime.DefaultHTTPErrorHandler(ctx, mux, m, w, req, err)
-				return
-			}
-			if redirectPath != referer.Path {
-				redirectPath = fmt.Sprintf("%s?redirect_url=%s", redirectPath, referer.Path)
-			}
+	if isBrowser(req.Header) { // Redirect if it's the browser (non-XHR).
+		if s, ok := status.FromError(err); ok && s.Code() == codes.Unauthenticated {
+			redirectPath := fmt.Sprintf("/v1/authn/login?redirect_url=%s", url.QueryEscape(req.RequestURI))
+			http.Redirect(w, req, redirectPath, http.StatusFound)
```

Comment:
This might not be a concern since the response forwarder might prevent this edge case but just in case, does `Redirect` set a default value for the `Accept` header similar to how it sets a default `Content-Type` that might cause an infinite redirect loop?

Reply:
The `Accept` header is a request header, and `Redirect` is setting response headers (`Location` and `Content-Type`). Not sure I follow exactly, but the `Content-Type` header shouldn't influence the follow up request after redirect headers at all.